gammapy/modeling/models/temporal.py

```diff
@@ -454,97 +445,145 @@ class LightCurveTemplateTemporalModel(TemporalModel):
+        m = RegionNDMap.create(region=None, axes=axes, meta=table.meta)
 
+        return cls(m, t_ref=t_ref)
+
+    @classmethod
+    def read(cls, filename, format="table"):
+        """Read a TemplateModel from disk
+
+        Parameters:
+
+        filename : str
+            Name of file to read
+        format : str
+            Format of the input file.
+            either "table" or "map"
 
-        Returns
-        -------
-        norm : array_like
-            Norm at the given times.
         """
-        return self._interpolator(time, ext=ext)
+        filename = str(make_path(filename))
+        if format == "table":
+            table = Table.read(filename)
+            cls.from_table(table)
 
-    def integral(self, t_min, t_max):
-        """Evaluate the integrated flux within the given time intervals
+        elif format == "map":
+            m = RegionNDMap.read(filename)
+            t_ref = time_ref_from_dict(m.meta)
+            return cls(m, t_ref=t_ref)
 
-        Parameters
-        ----------
-        t_min: `~astropy.time.Time`
-            Start times of observation
-        t_max: `~astropy.time.Time`
-            Stop times of observation
-        Returns
-        -------
-        norm: The model integrated flux
+        else:
+            raise ValueError(f"Not a valid format: '{format}', choose from: {'table', 'map'}")
+
+    def to_table(self, format="map"):
+        """Convert model to an astropy table"""
+        columns = [_.upper() for _ in table.colnames]
+        if "ENERGY" in columns:
+            raise NotImplemented("Currently not supported for Energy Dependent Models")
+        table = Table(data=self.map.quantity, names=columns,
+                      meta=self.map.meta)
+        return table
+
+    def write(self, filename, format="table", overwrite=False):
+        """Write a model to disk as per the specified format
+
+        Parameters:
+            filename : str
+                name of output file
+            format : str, either "table" or "map"
+                if format is table, it is serialised as an astropy Table
+                if map, then it is serialised as a RegionNDMap
+            overwrite : bool
+                Overwrite file on disk if present
         """
 
-        n1 = self._interpolator.antiderivative()(t_max.mjd)
-        n2 = self._interpolator.antiderivative()(t_min.mjd)
-        return u.Quantity(n1 - n2, "day") / self.time_sum(t_min, t_max)
+        if self.filename is None:
+            raise IOError("Missing filename")
+
+        if format == "table":
+            table = self.to_table()
+            table.write(filename, overwrite=overwrite)
+
+        elif format == "map":
+            self.map.write(filename, overwrite=overwrite)
+
+        else:
+            raise ValueError("Not a valid format")
+
+    def evaluate(self, time, energy=None):
+        """Evaluate the model at given coordinates."""
+        coords = {"time": time.mjd}
```

Comment:
I think the time needs to be converted into a time relative to the reference time parameter.